sdks/java/io/tika/src/main/java/org/apache/beam/sdk/io/tika/TikaIO.java

```diff
@@ -0,0 +1,307 @@
+ *
+ * <p>Example:
+ *
+ * <pre>{@code
+ * Pipeline p = ...;
+ *
+ * // A simple Read of a local PDF file (only runs locally):
+ * PCollection<String> content = p.apply(TikaInput.from("/local/path/to/file.pdf"));
+ * }</pre>
+ */
+public class TikaIO {
+
+  /**
+   * A {@link PTransform} that parses one or more files and returns a bounded {@link PCollection}
+   * containing one element for each sequence of characters reported by Apache Tika SAX Parser.
+   */
+   public static Read read() {
+     return new AutoValue_TikaIO_Read.Builder()
+        .setQueuePollTime(Read.DEFAULT_QUEUE_POLL_TIME)
+        .setQueueMaxPollTime(Read.DEFAULT_QUEUE_MAX_POLL_TIME)
+        .build();
+   }
+
+   /** Implementation of {@link #read}. */
+  @AutoValue
+  public abstract static class Read extends PTransform<PBegin, PCollection<String>> {
+    private static final long serialVersionUID = 2198301984784351829L;
+    public static final long DEFAULT_QUEUE_POLL_TIME = 50L;
+    public static final long DEFAULT_QUEUE_MAX_POLL_TIME = 3000L;
+
+    @Nullable abstract ValueProvider<String> getFilepattern();
+    @Nullable abstract ValueProvider<String> getTikaConfigPath();
+    @Nullable abstract Metadata getInputMetadata();
+    @Nullable abstract Boolean getReadOutputMetadata();
+    @Nullable abstract Long getQueuePollTime();
+    @Nullable abstract Long getQueueMaxPollTime();
+    @Nullable abstract Integer getMinimumTextLength();
+    @Nullable abstract Boolean getParseSynchronously();
+
+    abstract Builder toBuilder();
+
+    @AutoValue.Builder
+    abstract static class Builder {
+      abstract Builder setFilepattern(ValueProvider<String> filepattern);
+      abstract Builder setTikaConfigPath(ValueProvider<String> tikaConfigPath);
+      abstract Builder setInputMetadata(Metadata metadata);
+      abstract Builder setReadOutputMetadata(Boolean value);
+      abstract Builder setQueuePollTime(Long value);
+      abstract Builder setQueueMaxPollTime(Long value);
+      abstract Builder setMinimumTextLength(Integer value);
+      abstract Builder setParseSynchronously(Boolean value);
+
+      abstract Read build();
+    }
+
+    /**
+     * A {@link PTransform} that parses one or more files with the given filename
+     * or filename pattern and returns a bounded {@link PCollection} containing
+     * one element for each sequence of characters reported by Apache Tika SAX Parser.
+     *
+     * <p>Filepattern can be a local path (if running locally), or a Google Cloud Storage
+     * filename or filename pattern of the form {@code "gs://<bucket>/<filepath>"}
+     * (if running locally or using remote execution service).
+     *
+     * <p>Standard <a href="http://docs.oracle.com/javase/tutorial/essential/io/find.html" >Java
+     * Filesystem glob patterns</a> ("*", "?", "[..]") are supported.
+     */
+    public Read from(String filepattern) {
+      checkNotNull(filepattern, "Filepattern cannot be empty.");
```

Comment:
Use of checkNotNull is against the current best practices for validation https://beam.apache.org/contribute/ptransform-style-guide/#validation